aea/cli/generate.py

```diff
@@ -0,0 +1,124 @@
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Implementation of the 'aea generate' subcommand."""
+
+import os
+import shutil
+import sys
+from pathlib import Path
+
+import click
+from jsonschema import ValidationError
+import yaml
+
+# from aea import AEA_DIR
+from aea.cli.common import Context, pass_ctx, logger, _try_to_load_agent_config
+from aea.configurations.base import PublicId, DEFAULT_AEA_CONFIG_FILE
+from aea.protocols.generator import ProtocolTemplate, ProtocolGenerator, ProtocolSpecificationParseError
+
+# these variables are being used dynamically
+from aea.configurations.base import DEFAULT_CONNECTION_CONFIG_FILE, DEFAULT_PROTOCOL_CONFIG_FILE, DEFAULT_SKILL_CONFIG_FILE  # noqa: F401
+
+
+@click.group()
+@pass_ctx
+def generate(ctx: Context):
+    """Generate a resource for the agent."""
+    _try_to_load_agent_config(ctx)
+
+
+def _generate_item(ctx: Context, item_type, item_name, specification_path):
+    """Generate an item based on a specification and add it to the configuration file and agent."""
+    existing_id_list = getattr(ctx.agent_config, "{}s".format(item_type))
+    existing_item_list = [public_id.name for public_id in existing_id_list]
+
+    # loader = getattr(ctx, "{}_loader".format(item_type))
+    # default_config_filename = globals()["DEFAULT_{}_CONFIG_FILE".format(item_type.upper())]
+
+    item_type_plural = item_type + "s"
+
+    protocol_template = ProtocolTemplate(specification_path)
+
+    try:
+        protocol_template.load()
+    except (yaml.YAMLError, ProtocolSpecificationParseError) as e:
+        print(str(e))
+        print("Load error, exiting now!")
+        exit(1)
+
+    # check if we already have an item with the same name
+    logger.debug("{} already supported by the agent: {}".format(item_type_plural, existing_item_list))
+    protocol_name = protocol_template.name
+    if protocol_name in existing_item_list:
+        logger.error("A {} with name '{}' already exists. Aborting...".format(item_type, protocol_name))
+        sys.exit(1)
+
+    # import pdb; pdb.set_trace()
+
+    try:
+        agent_name = ctx.agent_config.agent_name
+        logger.info("Generating {} '{}' and adding it to the agent '{}'...".format(item_type, protocol_name, agent_name))
+
+        # output_path = Path(os.path.join(ctx.cwd, item_type_plural))
+        output_path = os.path.join(ctx.cwd, item_type_plural)
+        protocol_generator = ProtocolGenerator(protocol_template, output_path)
+        protocol_generator.generate()
+
+        Path(item_type_plural).mkdir(exist_ok=True)
```

Comment:
Is this still required here.. I assume the protocol_generator will already have created the directory?

Reply:
Also, remove unused code below

Reply:
Done